### DISCUSSION TITLE
How to combine storage and database?

### DISCUSSION THREAD

**Discussion by yanivoxy** [supabase/supabase#10720] [Questions]
I have images and videos for products in storage and names for them in a database how can i combine both of them?

**Comment by GaryAustin1** [comment]
Can you give more information about what you mean by combine them?   
By names I assume you mean pathnames used to upload the files into storage.

**Reply by GaryAustin1** [reply]
You should normally not start new answers to discuss an answer...

You normally get the pathname from your upload operation and then save it to your table as the "pointer" to the file in storage.  
To get the file later, you get the pathname from your table and as long as you know the bucket is always the same just use those to build a URL or call a function to get a public or signed URL.  You could also store the entire public URL in your table, but I recommend not doing that because if you move instances, you have to update all those URLs.

**Comment by yanivoxy** [comment]
Correct. I need the pathname of the image with the corresponding title from
the db when uploaded. They have to have the same id to get them both?
